### PR TITLE
Cherrypicks 3.3.5a -> master

### PR DESCRIPTION
**Pull request by funjoker**
**Changes proposed:**

-  Cherrypicks

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by qxpf666** [comment]
that is good

**Comment by funjoker** [comment]
Last Commit for this PR.
Keeping it small.

**Comment by funjoker** [comment]
Someone knows which Commit in 3.3.5 added Orientation to spell target position?

**Comment by msoky** [comment]
is this mergable?

File: src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp [hunk 1]
```diff
@@ -1091,8 +1094,14 @@ class spell_sindragosa_s_fury : public SpellScriptLoader
                 }
             }
 
-            void CountTargets(std::list<WorldObject*>& targets)
+            void FilterTargets(std::list<WorldObject*>& targets)
             {
+                targets.remove_if([](WorldObject* obj) -> bool
+                {
+                    // SPELL_ATTR3_ONLY_TARGET_PLAYERS present on the spell, we can safely cast to Player
+                    return obj->ToPlayer()->IsGameMaster();
```
Comment (Keader) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp]: if i remember right, has a crash here.
Players corpse will cause crash.
Get current 3.3.5a version :)

**Comment by illfated** [comment]
@msoky : Not to meaning to dismiss your question, the file conflicts need to be resolved first.
After that, some other issues need to be resolved too, like the one Keader mentioned above.

**Pull request merged**